kv/db_test.go

```diff
@@ -222,8 +222,7 @@ func TestKVDBTransaction(t *testing.T) {
 			t.Errorf("expected value %q; got %q", value, gr.ValueBytes())
 		}
 		return nil
-	})
-	if err != nil {
+	}); err != nil {
```

Comment:
FWIW it's not exclusive to embedded closures; the same thing happens with certain function calls.

I think it's worth it for the scope narrowing, personally.